### PR TITLE
Add documentation to sinkbinding modes

### PR DESCRIPTION
**Pull request by lberk**
## Proposed Changes 

- Document the sinkbinding modes and their side effects in relation to namespaces and labels

### CODE REVIEW

File: docs/eventing/sources/sinkbinding.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ For example, `Deployment`, `Job`, `DaemonSet`, or `StatefulSet` objects, or Knat
 
 Sink binding injects environment variables into the `PodTemplateSpec` of the event sink, so that the application code does not need to interact directly with the Kubernetes API to locate the event destination.
 
+The `SinkBinding` operates in one of two modes; `Inclusion` or `Exclusion`.  This can be set by modifying the `SINK_BINDING_SELECTION_MODE` of the `eventing-webhook` deployment accordingly.
+
+The selection mode determines the default scope of the webhook. By default, the upstream webhook is set to `exclusion` mode, which means any `Namespace` that does **not** have the label `bindings.knative.dev/exclude: true` will be subject to mutation evalutation.  Conversely, if `SINK_BINDING_SELECTION_MODE` is set to `inclusion`, only the resources in a namespaces labelled with `bindings.knative.dev/include: true` will be considered.  In `inclusion` mode, any `sinkbinding` resource created will automatically label the `subject`'s namespace with `bindings.knative.dev/include: true` for inclusion in the potential environment variable inclusions.
```
Comment (abrennan89) [inline docs/eventing/sources/sinkbinding.md:23]:
```suggestion
Sink binding operates in one of two modes; `Inclusion` or `Exclusion`.
You can set the mode by modifying the `SINK_BINDING_SELECTION_MODE` of the `eventing-webhook` deployment accordingly. The mode determines the default scope of the webhook.

By default, the webhook is set to `exclusion` mode, which means that any namespace that does not have the label `bindings.knative.dev/exclude: true` will be subject to mutation evalutation.

If `SINK_BINDING_SELECTION_MODE` is set to `inclusion`, only the resources in a namespace labelled with `bindings.knative.dev/include: true` will be considered.  In `inclusion` mode, any SinkBinding resource created will automatically label the `subject` namespace with `bindings.knative.dev/include: true` for inclusion in the potential environment variable inclusions.
```

**Review by abrennan89 — CHANGES_REQUESTED** [review]
@lberk just some edits :slightly_smiling_face:

**Comment by lberk** [comment]
awesome, thanks @abrennan89! (though I think that means I just annoyed cla-bot and you might have to consent to me having used your edit?)

**Comment by lberk** [comment]
@google-prow-robot I consent.

**Comment by lberk** [comment]
@googlebot I consent.

**Comment by abrennan89** [comment]
@googlebot I consent

**Comment by abrennan89** [comment]
@lberk I think repushing the PR also works :slightly_smiling_face:

**Review by abrennan89 — APPROVED** [review]
/lgtm

**Comment by abrennan89** [comment]
Idk why this isn't working :confounded: @RichieEscarez can you please take a look in case it needs to be added manually? Are we missing something here?

**Comment by RichieEscarez** [comment]
@abrennan89 are you on a different machine or maybe change your local config (ie [private account settings](https://knative.dev/community/contributing/contributing/#contributor-license-agreements))? The CLA is failing because of this email: `40172997+abrennan89@users.noreply.github.com`.

**Comment by abrennan89** [comment]
@RichieEscarez yes and no, my machine died so it had to be resurrected and reinstalled everything etc. Any idea how to fix that?

**Comment by lberk** [comment]
@googlebot I consent.

@abrennan89 I edited the commit message to fix the email from github's auto-commit, sorry about that!

**Comment by abrennan89** [comment]
@googlebot I consent.

**Comment by abrennan89** [comment]
/lgtm

Please cla-bot :joy:

**Pull request merged**